docs/source/index.rst

```diff
@@ -64,6 +64,7 @@ Features described in this documentation are classified by release status:
    torch.distributed.fsdp <fsdp>
    torch.distributed.optim <distributed.optim>
    torch.distributions <distributions>
+   utils/inference/run
```

Comment:
lets follow the conventions here:

1. Create a main rst page for `docs/source/utils/run.rst` and give it the title `<torchrun>`. (e.g. `torchrun <utils.run>`)
2. In `utils/run.rst` explain at the high level what `torchrun` is doing, then link the two types of launchers from it (distributed.elastic.run and utils.inference.run)